I am trying to run `pnpm lint:fix` which seems to give a different output compared to the errors in CI. Is there something missing in my local setup?

```
$ /home/karthikeyan/.local/share/pnpm/pnpm lint:fix

> ui@0.0.0 lint:fix /home/karthikeyan/stuff/python/airflow/airflow/ui
> eslint --fix && tsc --p tsconfig.app.json


/home/karthikeyan/stuff/python/airflow/airflow/ui/src/App.test.tsx
  35:20  warning  Use `undefined` instead of `null`  unicorn/no-null
  43:21  warning  Use `undefined` instead of `null`  unicorn/no-null
  45:21  warning  Use `undefined` instead of `null`  unicorn/no-null
  54:18  warning  Use `undefined` instead of `null`  unicorn/no-null
  55:23  warning  Use `undefined` instead of `null`  unicorn/no-null
  64:20  warning  Use `undefined` instead of `null`  unicorn/no-null
  72:21  warning  Use `undefined` instead of `null`  unicorn/no-null
  74:21  warning  Use `undefined` instead of `null`  unicorn/no-null
  83:18  warning  Use `undefined` instead of `null`  unicorn/no-null
  84:23  warning  Use `undefined` instead of `null`  unicorn/no-null

/home/karthikeyan/stuff/python/airflow/airflow/ui/src/pages/DagsList/DagsFilters.tsx
  30:28  warning  Arrow function has too many statements (12). Maximum allowed is 10  max-statements

/home/karthikeyan/stuff/python/airflow/airflow/ui/src/utils/index.tsx
  20:1   error  Expected a function expression                                                                             func-style
  27:10  error  Unexpected nullable string value in conditional. Please handle the nullish/empty cases explicitly          @typescript-eslint/strict-boolean-expressions
  27:39  error  Prefer using nullish coalescing operator (`??`) instead of a logical or (`||`), as it is a safer operator  @typescript-eslint/prefer-nullish-coalescing

✖ 14 problems (3 errors, 11 warnings)

 ELIFECYCLE  Command failed with exit code 1.

```

--- a/airflow/ui/src/pages/DagsList/DagsFilters.tsx
+++ b/airflow/ui/src/pages/DagsList/DagsFilters.tsx
@@ -25,11 +25,17 @@ import { useTableURLState } from "src/components/DataTable/useTableUrlState";
 import { QuickFilterButton } from "src/components/QuickFilterButton";
 
 const PAUSED_PARAM = "paused";
+const STATE_PARAM = "lastrun";
 
 export const DagsFilters = () => {
   const [searchParams, setSearchParams] = useSearchParams();
 
   const showPaused = searchParams.get(PAUSED_PARAM);
+  const state = searchParams.get(STATE_PARAM);
+  const isAll = state === null || state === "all";
+  const isRunning = Boolean(state) && state === "running";
+  const isFailed = Boolean(state) && state === "failed";
+  const isSuccess = Boolean(state) && state === "success";
 
   const { setTableURLState, tableURLState } = useTableURLState();
   const { pagination, sorting } = tableURLState;
@@ -51,6 +57,19 @@ export const DagsFilters = () => {
       [pagination, searchParams, setSearchParams, setTableURLState, sorting],
     );
 
+  const handleStateChange: React.ChangeEventHandler<HTMLSelectElement> =
+    useCallback(
+      ({ target: { value } }) => {
+        if (value === "all") {
+          searchParams.delete(STATE_PARAM);
+        } else {
+          searchParams.set(STATE_PARAM, value);
+        }
+        setSearchParams(searchParams);
+      },
+      [searchParams, setSearchParams],
+    );
+
   return (
     <HStack justifyContent="space-between">
       <HStack spacing={4}>
@@ -59,10 +78,34 @@ export const DagsFilters = () => {
             State:
           </Text>
           <HStack>
-            <QuickFilterButton isActive>All</QuickFilterButton>
-            <QuickFilterButton isDisabled>Failed</QuickFilterButton>
-            <QuickFilterButton isDisabled>Running</QuickFilterButton>
-            <QuickFilterButton isDisabled>Successful</QuickFilterButton>
+            <QuickFilterButton
+              isActive={isAll}
+              onClick={handleStateChange}
+              value="all"
+            >
+              All
+            </QuickFilterButton>
+            <QuickFilterButton
+              isActive={isFailed}
+              onClick={handleStateChange}
+              value="failed"
+            >
+              Failed
+            </QuickFilterButton>
+            <QuickFilterButton
+              isActive={isRunning}
+              onClick={handleStateChange}
+              value="running"
+            >
+              Running
+            </QuickFilterButton>
+            <QuickFilterButton
+              isActive={isSuccess}
+              onClick={handleStateChange}
+              value="success"
+            >
+              Successful
+            </QuickFilterButton>
           </HStack>
         </Box>
         <Box>

--- a/airflow/ui/src/pages/DagsList/DagsFilters.tsx
+++ b/airflow/ui/src/pages/DagsList/DagsFilters.tsx
@@ -57,13 +57,13 @@ export const DagsFilters = () => {
       [pagination, searchParams, setSearchParams, setTableURLState, sorting],
     );
 
-  const handleStateChange: React.ChangeEventHandler<HTMLSelectElement> =
+  const handleStateChange: React.MouseEventHandler<HTMLButtonElement> =
     useCallback(
       ({ target: { value } }) => {
         if (value === "all") {
           searchParams.delete(STATE_PARAM);
         } else {
-          searchParams.set(STATE_PARAM, value);
+          searchParams.set(STATE_PARAM, value as string);
         }
         setSearchParams(searchParams);
       },

--- a/airflow/ui/src/pages/DagsList/DagsFilters.tsx
+++ b/airflow/ui/src/pages/DagsList/DagsFilters.tsx
@@ -25,17 +25,17 @@ import { useTableURLState } from "src/components/DataTable/useTableUrlState";
 import { QuickFilterButton } from "src/components/QuickFilterButton";
 
 const PAUSED_PARAM = "paused";
-const STATE_PARAM = "lastrun";
+const STATE_PARAM = "last_dag_run_state";
 
 export const DagsFilters = () => {
   const [searchParams, setSearchParams] = useSearchParams();
 
   const showPaused = searchParams.get(PAUSED_PARAM);
   const state = searchParams.get(STATE_PARAM);
-  const isAll = state === null || state === "all";
-  const isRunning = Boolean(state) && state === "running";
-  const isFailed = Boolean(state) && state === "failed";
-  const isSuccess = Boolean(state) && state === "success";
+  const isAll = state === null;
+  const isRunning = state === "running";
+  const isFailed = state === "failed";
+  const isSuccess = state === "success";
 
   const { setTableURLState, tableURLState } = useTableURLState();
   const { pagination, sorting } = tableURLState;
@@ -59,17 +59,19 @@ export const DagsFilters = () => {
 
   const handleStateChange: React.MouseEventHandler<HTMLButtonElement> =
     useCallback(
-      ({ target }) => {
-        const { value } = target as HTMLButtonElement;
-
+      ({ currentTarget: { value } }) => {
         if (value === "all") {
           searchParams.delete(STATE_PARAM);
         } else {
           searchParams.set(STATE_PARAM, value);
         }
         setSearchParams(searchParams);
+        setTableURLState({
+          pagination: { ...pagination, pageIndex: 0 },
+          sorting,
+        });
       },
-      [searchParams, setSearchParams],
+      [pagination, searchParams, setSearchParams, setTableURLState, sorting],
     );
 
   return (

--- a/airflow/ui/src/pages/DagsList/DagsList.tsx
+++ b/airflow/ui/src/pages/DagsList/DagsList.tsx
@@ -97,12 +97,14 @@ const cardDef: CardDef<DAGResponse> = {
 };
 
 const PAUSED_PARAM = "paused";
+const STATE_PARAM = "lastrun";
 
 export const DagsList = () => {
   const [searchParams] = useSearchParams();
   const [display, setDisplay] = useState<"card" | "table">("card");
 
   const showPaused = searchParams.get(PAUSED_PARAM);
+  const lastDagRunState = searchParams.get(STATE_PARAM);
 
   const { setTableURLState, tableURLState } = useTableURLState();
   const { pagination, sorting } = tableURLState;
@@ -112,6 +114,7 @@ export const DagsList = () => {
   const orderBy = sort ? `${sort.desc ? "-" : ""}${sort.id}` : undefined;
 
   const { data, isFetching, isLoading } = useDagServiceGetDags({
+    lastDagRunState,
     limit: pagination.pageSize,
     offset: pagination.pageIndex * pagination.pageSize,
     onlyActive: true,

--- a/airflow/ui/src/pages/DagsList/DagsList.tsx
+++ b/airflow/ui/src/pages/DagsList/DagsList.tsx
@@ -97,7 +97,7 @@ const cardDef: CardDef<DAGResponse> = {
 };
 
 const PAUSED_PARAM = "paused";
-const STATE_PARAM = "lastrun";
+const STATE_PARAM = "last_dag_run_state";
 
 export const DagsList = () => {
   const [searchParams] = useSearchParams();

